### PR TITLE
Include GHC 9.0.2 in CI, remove unnecessary CPP

### PR DESCRIPTION
**Pull request by jhrcek**

### CODE REVIEW

File: .github/workflows/haskell-ci.yml [hunk 1]
```diff
@@ -11,113 +11,108 @@ jobs:
...
+            ghc: 8.10.7
           - os: windows-latest
             ghc: 8.6.5
           - os: windows-latest
             ghc: 8.8.4
+          - os: windows-latest
+            ghc: 8.10.7
 
     steps:
-    - uses: actions/checkout@v2
```
Comment (jhrcek) [inline .github/workflows/haskell-ci.yml:37]: These changes come from yaml formatter

File: .github/workflows/haskell-ci.yml [hunk 2]
```diff
@@ -11,113 +11,108 @@ jobs:
...
       matrix:
-        os: [ubuntu-latest, macOS-latest, windows-latest]
-        cabal: ["3.2"]
+        os:
+          - ubuntu-latest
+          - macOS-latest
+          - windows-latest
+        cabal: [3.6]
         ghc:
-          - "8.4.4"
```
Comment (turboMaCk) [inline .github/workflows/haskell-ci.yml:17]: maybe you can already put that we're dropping this into change log so we don't forget https://github.com/Holmusk/elm-street/blob/master/CHANGELOG.md

**Review by turboMaCk — APPROVED** [review]
LGTM

File: .github/workflows/haskell-ci.yml (hunk 2 above)
Comment (jhrcek) [reply .github/workflows/haskell-ci.yml:17]: Good point, updated in the next commit. Also mentioned other not-yet-released changes.

**Pull request merged**